### DISCUSSION TITLE
Usage of shop-trace

### DISCUSSION THREAD

**Discussion by ko56** [shop-planner/shop3#162] [Q&A]
I cannot get `shop-trace` to work as I think the documentation says it should.
For example, I have an operator `(:op (!!read-planner-params ?fname)`. This is what happens
```
shop-user> (shop-trace !!read-planner-params)
warning: Ignoring invalid shop-trace argument !!read-planner-params
nil
shop-user> 
```
According to the manual, this should work, it is an instance of `(shop-trace <item>)`. 
But then I look at the code of `shop-trace-1` in `io/debugging.lisp`, and, as far as I understand this code, it produces the error message correctly.  
Am I calling `shop-trace` incorrectly, or is the code incomplete?

**Comment by rpgoldman** [comment]
The call is not correct, but the documentation is probably not at all clear. As far as I can tell, the shop trace code was never fully completed, and I'm not sure it works properly on all the arguments it is advertised to.

I *think* what you need to do is `(push '!!read-planner-params *traced-tasks*)`.  It looks like there was once a `*traced-operators*` list, but that looks like it was removed before I started working on the SHOP source code.

Also, one would _like_ the traced tasks and operators to print information only for those specific items (because something like `(shop-trace :tasks)` often produces way too much output).  Unfortunately, instead of just _tracing_ those items cause a breakpoint when matching items are applied.  This is all very messy, and I wish I could fix it up, but I don't have the time to do it right now.

Having "break" and "trace" be distinct would be an improvement, as would be printing information only about *some* tasks, goals, methods, etc.

**Reply by ko56** [reply]
When you have some time, please look at #164.
